src/ezloCap.c

```diff
@@ -0,0 +1,277 @@
+}
+
+void NPC4E(Entity* this) {
+    if (this->action == 0) {
+        this->action = 1;
+        this->spriteSettings.b.draw = 4;
+        this->boundingBox = &gUnk_080FD170;
+        sub_0807DD50(this);
+    } else {
+        sub_0807DD94(this, 0);
+    }
+}
+
+NONMATCH("asm/non_matching/ezloCap/sub_0806DA04.inc", void sub_0806DA04(Entity* this, u32 param_2)) {
+    u8 direction =
+        sub_080045D4(this->x.HALF.HI, this->y.HALF.HI, (int)gPlayerEntity.x.HALF.HI, (int)gPlayerEntity.y.HALF.HI);
+    this->direction = direction;
+
+    this->animationState =
+        (this->animationState & 0x80) |
+        gUnk_08114144[(((direction) << 0x18) >> 0x19)]; // somehow the cast from u8 to u32 of direction is an issue here
+}
+END_NONMATCH
+
+void sub_0806DA1C(Entity* this, u32* param_2) {
+    sub_0808091C((ScreenTransitionData*)(gUnk_081141A4)[param_2[1]], (u32)(gUnk_081141E4[param_2[1]]));
+}
+
+// Returns the kingstone id?
+u8 sub_0806DA3C(Entity* this) {
+    u32 result;
+
+    switch (this->entityType.form) {
+        default:
+            result = 0;
+            break;
+        case 1:
+            result = 1;
+            break;
+        case 2:
+            result = 2;
+            break;
+        case 3:
+            result = 3;
+            break;
+        case 4:
+            result = 4;
+            break;
+        case 5:
+            result = 5;
+            break;
+        case 11:
+            result = 9;
+            break;
+        case 6:
+        case 7:
+        case 8:
+        case 9:
+        case 10:
+            result = sub_0801E99C(this);
+            break;
+    }
+    return result;
+}
+
+// Check whether a kingstone fusion is possible and store the result somewhere in param_2?
+void sub_0806DAAC(Entity* this, u32 param_2) {
+    u32 uVar1;
+    uVar1 = CheckKinstoneFused(sub_0806DA3C(this));
+    *(u32*)(param_2 + 0x14) = uVar1;
+    gUnk_02033280.unk_07 |= 1;
+}
+
+// maybe actually execute the keystone fusion?
+void sub_0806DAD0(Entity* this) {
+    sub_08078790(this, sub_0806DA3C(this));
+}
+
+void sub_0806DAE8(Entity* this) {
+    switch (this->entityType.form - 1) {
+        case 0:
+            SetLocalFlag(0xee);
+            break;
+        case 1:
+            SetLocalFlag(0xef);
+            break;
+        case 2:
+            SetLocalFlag(0xf0);
+            break;
+        case 3:
+            SetLocalFlag(0xf1);
+            break;
+        case 4:
+            SetLocalFlag(0xf2);
+            break;
+    }
+
+    PlaySFX(0xcd);
+}
+
+// Unknown param_2 struct
+void sub_0806DB44(Entity* this, u32* param_2) {
+    param_2[5] = 0;
+
+    if (CheckLocalFlag(0xee) != 0 && CheckLocalFlag(0xef) != 0 && CheckLocalFlag(0xf0) != 0 &&
+        CheckLocalFlag(0xf1) != 0 && CheckLocalFlag(0xf2) != 0) {
+        param_2[5] = 1;
+    }
+}
+
+void sub_0806DB84(
+    Entity* this,
+    u8 unused) { // The unused param just had to be added, so that a mov r1, #0 in NPC4E_Fusion is matching correctly
```

Comment:
These linebreaks feel weird to me - I'd move the comment to the line above to avoid overly long lines instead
```suggestion
// The unused param just had to be added, so that a mov r1, #0 in NPC4E_Fusion is matching correctly
void sub_0806DB84(Entity* this, u8 unused) {
```